src/bogart/AS_BAT_MarkRepeatReads.C

```diff
@@ -341,29 +279,70 @@ isInside(int32 lo, int32 coord, int32 hi) {
 }
 
 
-uint32
-findThickestPrevRead(Unitig *tig, uint32 fi, int32 rMin, int32 rMax) {
-  ufNode     *rdA       = &tig->ufpath[fi];
-  uint32      rdAid     = rdA->ident;
-  bool        rdAfwd    = rdA->position.isForward();
-  int32       rdAlo     = rdA->position.min();
-  int32       rdAhi     = rdA->position.max();
+//
+//  General comments on findThickestPrevRead() (similar for findThickestNextRead())
+//  that were distracting if inlined in the code below.
+//
+//
+//  Decide if the repeat region is to the left or right of this read.
+//  Then ignore this thickest read if it doesn't fall in the repeat correctly.
+//
+//  A prev that ends inside the repeat, and so could be confused:
+//
+//              ----[rrrrrrrrrrrrrrrrrrrrrrrrrrrrr]------------------
+//                                   ----------------------rdA
+//                    rdB------------------
+//
+//  A prev that ends outside the repeat, can't be confused.
+//
+//              ----[rrrrrrrrrrrrrrrrrrrrrrrrrrrrr]------------------
+//                                   ----------------------rdA
+//                    rdB----------------------------
+//
+//  If the high coordinate isn't inside the repeat, we don't care;
+//  rdB will be anchored to unique sequence.
+//
+//  For finding the previous read:
+//
+//        --------[repeat]]]--------
+//                   -------------         READ WE'RE SEARCHING FROM
+//                 ---------u              DON'T CARE, anchored in unique (u)
+//          -----                          DON'T CARE, shouldn't happen
+//
+//  For the next read:
+//
+//        ------[[[repeat]----------
+//       -------------                     READ WE'RE SEARCHING FROM
+//             u---------                  DON'T CARE, anchored in unique (u)
+//                          ------         DON'T CARE, shouldn't happen
+//
+//  The repeat is extended a bit so that the read will have some useful bit
+//  of unique anchoring.
+//
+
 
-  uint32   bestId  = 0;
-  uint32   bestIdx = 0;
+ufNode *
+findThickestPrevRead(Unitig *tig, uint32 fi, int32 rMin, int32 rMax, char *logMsg) {
+  ufNode  *rdA     = &tig->ufpath[fi];
+  uint32   rdAid   = rdA->ident;
+  bool     rdAfwd  = rdA->position.isForward();
+  int32    rdAlo   = rdA->position.min();
+  int32    rdAhi   = rdA->position.max();
+
+  uint32   bestIdx = UINT32_MAX;
   uint32   bestLen = 0;
   uint32   olapMin = 0;
-  uint32   olapMax = 0;;
+  uint32   olapMax = 0;
 
-  //  If rdA begins before rMin, we don't need to find the previous read.
-  //  it's not confused.
+  //  If rdA begins before the repeat, it can't be confused; return a null
+  //  previous read.
 
   if (rdAlo < rMin)
-    return(UINT32_MAX);
+    return(nullptr);
 
   //  Otherwise, search for the previous best read.
 
-  for (int32 pi=fi-1; pi>0; pi--) {
+  for (int32 pi=fi-1; pi >= 0; pi--) {
     ufNode  *rdB   = &tig->ufpath[pi];
```

Comment:
Why is it a pointer and not const reference?

Reply:
I guess because rdB can sometimes be nullptr, right?

Reply:
I'm just used to and comfortable with pointers.  Updating const-ness will be a big project.

Reply:
Bri, while I understand that it is a matter of a background, I think we should all strive to follow the good practices, especially when there is virtually no downsides to them. Using const references instead of pointers (if easy to do, usually meaning that value can not be nullptr) makes code more readable and safer. They have also been in c++ forever ;) Using const references for local variables in many cases like here is not propagating anywhere and I think is independent from a bigger project of introducing const modifiers throughout the codebase.